Задание 1                        
Скачайте с сайта https://grouplens.org/datasets/movielens/ датасет любого размера.                        
Определите какому фильму было выставлено больше всего оценок 5.0.

In [3]:
import pandas as pd

In [8]:
%%time
ratings = pd.read_csv('ratings.csv')
movies = pd.read_csv('movies.csv')
sorted_1 = ratings.sort_values('movieId', ascending = True)
#sorted_1.head(5)
joined = sorted_1.merge(movies, on = 'movieId', how = 'left')
#joined.head(5)
data = joined[joined.rating >= 5]
data['title'].value_counts().head(5)

Wall time: 2min 31s
Parser   : 135 ms


Shawshank Redemption, The (1994)             39553
Pulp Fiction (1994)                          32169
Forrest Gump (1994)                          25918
Star Wars: Episode IV - A New Hope (1977)    25804
Matrix, The (1999)                           25482
Name: title, dtype: int64

'Shawshank Redemption' - было выставлено максимальное количество оценок 5.

Задание 2                                                        
По данным файла power.csv посчитайте суммарное потребление стран Прибалтики (Латвия, Литва и Эстония)                           
категорий 4, 12 и 21 за период с 2005 по 2010 года. Не учитывайте в расчетах отрицательные значения quantity.

In [9]:
power = pd.read_csv('power.csv')
power = power[power.quantity > 0]
power = power[(power['country'] =='Latvia') | (power['country'] =='Lithuania') | (power['country'] =='Estonia')]
power = power[(power['category'] == 4) | (power['category'] == 12) | (power['category'] == 21)]
power = power[(power['year'] == 2005) | (power['year'] == 2010)]
#power.head(1200)
power['quantity'].sum()

76025.0

Суммарное потребление тран Прибалтики с учетом всех условий составило 76025.

Задание 3                          
Выберите страницу любого сайта с табличными данными.                           
Импортируйте таблицы в pandas dataframe.                          
Примеры страниц (необязательно брать именно эти):                         
https://fortrader.org/quotes
https://www.finanz.ru/valyuty/v-realnom-vremeni

In [13]:
data = pd.read_html('https://www.finanz.ru/valyuty/v-realnom-vremeni', encoding = 'utf8')
print(data)

[     0       1      2        3       4      5        6       7      8   \
0  ММВБ   3 683   0,7%   Nasdaq  13 402  -2,6%  Биткойн  54 920  -1,6%   
1   РТС   1 578   1,7%   S&P500   4 188  -1,0%    Нефть     678  -0,6%   
2   Dow  34 743  -0,1%  FTSE100   7 124  -0,1%   Золото   1 837   0,1%   

        9       10    11  
0  USD/RUB  743030  0,0%  
1  EUR/RUB  902575  0,1%  
2  EUR/USD   12150  0,1%  ,     Unnamed: 0              Рубль  Предложение  Спрос  Предыдущее закрытие  \
0          NaN            GBP/RUB      1046250      -              1049731   
1          NaN            EUR/RUB       899895      -               901875   
2          NaN            USD/RUB       739692      -               743383   
3          NaN            JPY/RUB        06810      -                06829   
4          NaN            CNY/RUB       115147      -               115853   
5          NaN            CHF/RUB       819664      -               824515   
6          NaN               Евро  Предложение 